### PR TITLE
prefer-node-remove: Only fix when expression is not used

### LINKED ISSUE #497: `prefer-node-remove`: prevent fixer when using return value
`prefer-node-append` currently checks for use of the return value since the replacement of `appendChild`, `append`, does not return a value.

`prefer-node-remove` should do the same, since the old `replaceChild` returned a value (the removed node) while `ChildNode.remove()` does not return a value.

### PR DESCRIPTION
**Pull request by brettz9**
Fixes #497

I also have a minor update and very minor refactoring included with this PR. Let me know if you need them separated out.

### CODE REVIEW

File: rules/prefer-node-remove.js [hunk 1]
```diff
@@ -64,11 +77,23 @@ const create = context => {
...
+					parent: grandparent
+				} = (parent || {});
+
+				let fix = fixer => fixer.replaceText(node, `${argumentName}.remove()`);
+
+				if (parent && ignoredParentTypes.includes(parent.type)) {
+					fix = undefined;
+				} else if (grandparent && ignoredGrandparentTypes.includes(grandparent.type)) {
+					fix = undefined;
+				}
```
Comment (fisker) [inline rules/prefer-node-remove.js]:
```suggestion
				if (
					(parent && ignoredParentTypes.includes(parent.type)) || 
					(grandparent && ignoredGrandparentTypes.includes(grandparent.type))
				) {
					fix = undefined;
				}
```

Comment (brettz9) [reply rules/prefer-node-remove.js]: Thanks! I've applied this, and also to the append rule.

**Comment by brettz9** [comment]
Applied, thanks.

Btw, I'm getting issues when running `npm run lint`, even before my changes. But `npm test` and `npm run integration` are working.

**Comment by fisker** [comment]
@brettz9 If they  are the some, can we add  a function maybe `isReturnValueUsed`(or a better name) in `/rules/utils`?

File: rules/prefer-node-append.js [hunk 2]
```diff
@@ -1,41 +1,18 @@
...
 
 			if (callee.type === 'MemberExpression' && getMethodName(callee) === 'appendChild') {
 				let fix = fixer => fixer.replaceText(callee.property, 'append');
 
-				if (parent && ignoredParentTypes.includes(parent.type)) {
-					fix = undefined;
-				}
-
-				if (grandparent && ignoredGrandparentTypes.includes(grandparent.type)) {
+				if (!isReturnValueUsed(node)) {
```
Comment (fisker) [inline rules/prefer-node-append.js]: Seems not right? If used, should has no fix, right?

**Comment by brettz9** [comment]
actually, on further reflection, I'm thinking `isValueUsed` would be better, since it is not necessarily part of a return. I've added a commit to change, though feel free to revert.

File: rules/prefer-node-append.js (hunk 2 above)
Comment (brettz9) [reply rules/prefer-node-append.js]: You're right--I inverted it by mistake. Plan to fix now.

Comment (brettz9) [reply rules/prefer-node-append.js]: Fixed.

File: rules/prefer-node-remove.js [hunk 3]
```diff
@@ -64,11 +65,17 @@ const create = context => {
 			)) {
 				const argumentName = getArgumentName(node.arguments);
 
+				let fix = fixer => fixer.replaceText(node, `${argumentName}.remove()`);
+
+				if (isValueUsed(node)) {
+					fix = undefined;
+				}
```
Comment (fisker) [inline rules/prefer-node-remove.js]:
```js
				const fix = isValueUsed(node) ? undefined : fixer => fixer.replaceText(node, `${argumentName}.remove()`);
```

Comment (fisker) [reply rules/prefer-node-remove.js]: And move it into `if` block, shall we?

Comment (brettz9) [reply rules/prefer-node-remove.js]: Ah, yes, all very good. Done.

**Comment by fisker** [comment]
Looks great, thank you very much.

I think `prefer-modern-dom-apis` also missing some check https://github.com/sindresorhus/eslint-plugin-unicorn/pull/362#discussion_r360750630 , If you're interested, maybe you can help, just let you know, no pressure.

I also notice, `prefer-modern-dom-apis` checked `nodeParentType === 'AssignmentExpression'` and we don't do this check here, not sure if is it needed.

**Comment by brettz9** [comment]
Yes, indeed, `nodeParentType === 'AssignmentExpression'` was needed. I've added that and some tests.

**Pull request merged**